### PR TITLE
EVA-1066 Tolerate the lack of HGVS_C and HGVS_T fields

### PR DESCRIPTION
**Pull request by jmmut**

### CODE REVIEW

File: dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/AssemblyCheckFilterProcessor.java [hunk 1]
```diff
@@ -45,7 +45,14 @@ public AssemblyCheckFilterProcessor(FastaSequenceReader fastaSequenceReader) {
      */
     @Override
     public SubSnpCoreFields process(SubSnpCoreFields subSnpCoreFields) throws Exception {
-        String referenceAllele = subSnpCoreFields.getReferenceInForwardStrand();
+        String referenceAllele;
+        try {
+            referenceAllele = subSnpCoreFields.getReferenceInForwardStrand();
+        } catch (IllegalArgumentException hgvsReferenceUndefined) {
+            logger.debug("Variant filtered out because reference allele is not defined: {} ({})", subSnpCoreFields,
+                         hgvsReferenceUndefined);
```
Comment (sundarvenkata-EBI) [inline dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/AssemblyCheckFilterProcessor.java]: does having hgvsReferenceUndefined as an argument automatically call hgvsReferenceUndefined.getMessage()?

Comment (jmmut) [reply dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/AssemblyCheckFilterProcessor.java]: the logger method takes `Object... object` which will replace the `{}` calling `object.toString()`. the `Throwable.toString()` does call the getMessage method.

File: dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/AssemblyCheckFilterProcessor.java [hunk 2]
```diff
@@ -45,7 +45,14 @@ public AssemblyCheckFilterProcessor(FastaSequenceReader fastaSequenceReader) {
      */
     @Override
     public SubSnpCoreFields process(SubSnpCoreFields subSnpCoreFields) throws Exception {
-        String referenceAllele = subSnpCoreFields.getReferenceInForwardStrand();
+        String referenceAllele;
+        try {
+            referenceAllele = subSnpCoreFields.getReferenceInForwardStrand();
+        } catch (IllegalArgumentException hgvsReferenceUndefined) {
```
Comment (cyenyxe) [inline dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/AssemblyCheckFilterProcessor.java]: Having to name the variable in some way to understand what is going on makes me think that we need a specific exception type here...

File: dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/UnambiguousAllelesFilterProcessor.java [hunk 3]
```diff
@@ -38,13 +38,29 @@ public UnambiguousAllelesFilterProcessor() {
...
-        Matcher referenceMatcher = pattern.matcher(subSnpCoreFields.getReferenceInForwardStrand());
+        String referenceInForwardStrand;
+        try {
+            referenceInForwardStrand = subSnpCoreFields.getReferenceInForwardStrand();
+        } catch (IllegalArgumentException hgvsReferenceUndefined) {
+            logger.debug("Variant filtered out because reference allele is not defined: {} ({})", subSnpCoreFields,
+                         hgvsReferenceUndefined);
+            return null;
+        }
+        Matcher referenceMatcher = pattern.matcher(referenceInForwardStrand);
```
Comment (cyenyxe) [inline dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/UnambiguousAllelesFilterProcessor.java]: Add a blank line before this one.

File: dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/UnambiguousAllelesFilterProcessor.java [hunk 4]
```diff
@@ -38,13 +38,29 @@ public UnambiguousAllelesFilterProcessor() {
...
-        Matcher alternateMatcher = pattern.matcher(subSnpCoreFields.getAlternateInForwardStrand());
+        String alternateInForwardStrand;
+        try {
+            alternateInForwardStrand = subSnpCoreFields.getAlternateInForwardStrand();
+        } catch (IllegalArgumentException hgvsAlternateUndefined) {
+            logger.debug("Variant filtered out because alternate allele is not defined: {} ({})", subSnpCoreFields,
+                         hgvsAlternateUndefined);
+            return null;
+        }
+        Matcher alternateMatcher = pattern.matcher(alternateInForwardStrand);
```
Comment (cyenyxe) [inline dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/UnambiguousAllelesFilterProcessor.java]: Add a blank line before this one.

**Pull request merged**